### DISCUSSION TITLE
Does the `.AppImage` embed the client?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [bpozdena/OneDriveGUI#137] [Q&A] [answered]
`.AppImage`s tend to include a darn lot (I believe, unnecessarily, but am unfamiliar with how to package the technology).

**Comment by bpozdena** [comment]
If you think somethings in the package are not necessary, which is definitely possible, please provide updated recipe for AppImage Builder. 
Ensure to provide test results from at least 10 major Linux distributions to confirm you have not removed a dependency that does not ship with that particular distro.

**Reply by roke-julian-lockhart** [reply]
@bpozdena, I've no desire to modify the package, although I am thankful for the advice. I'd just like an answer to the question, especially so that I can accurately answer [`discussions/138`](https://github.com/bpozdena/OneDriveGUI/discussions/138#discussioncomment-7603262).

**Reply by abraunegg** [reply]
@RokeJulianLockhart 
FYI - from reading a lot of your posts here and elsewhere, I find following your comments / posts / raising items then trying to link back - this all weaves at times a very difficult thread to follow at times. If you could perhaps tweak how you are raising items (issues, discussions and so on) that would be be potentially helpful.

> ... I'd just like an answer to the question, especially so that I can accurately answer ... 

So I am going to *assume* here the question you are asking is this: Does the .AppImage use the native client or embed its own? (the topic title)

To the best of my knowledge, the GUI as per this GitHub repository (https://github.com/bpozdena/OneDriveGUI) - regardless of appimage or not uses the 'onedrive' client that I develop and maintain: https://github.com/abraunegg/onedrive underneath the GUI options.

This is really clear in the readme (https://github.com/bpozdena/OneDriveGUI#readme) 

There is *no* native Microsoft provided Linux OneDrive Client.

**Reply by roke-julian-lockhart** [reply]
@abraunegg, regarding:

1.	1. > To the best of my knowledge, the GUI as per this GitHub repository ([`bpozdena/OneDriveGUI`](https://github.com/bpozdena/OneDriveGUI)) — regardless of AppImage or not, uses the '`onedrive`' client that I develop and maintain: [`abraunegg/onedrive`](https://github.com/abraunegg/onedrive) underneath the GUI options.

	1. > There is no native Microsoft provided Linux OneDrive Client.

	I'm not asking whether it uses a different codebase — I'm asking whether a compiled binary of [`OneDriveGUI`](https://github.com/bpozdena/OneDriveGUI.git) is statically linked inside [`OneDriveGUI-1.0.3-x86_64.AppImage`](https://github.com/bpozdena/OneDriveGUI/releases/download/v1.0.3/OneDriveGUI-1.0.3-x86_64.AppImage), or, whether, instead, [`OneDriveGUI-1.0.3-x86_64.AppImage`](https://github.com/bpozdena/OneDriveGUI/releases/download/v1.0.3/OneDriveGUI-1.0.3-x86_64.AppImage) utilises `$Env:PATH`, like most distribution-native packages do.

	I need to know in order to ascertain whether any configuration of [`openSUSE:/Factory/standard/x86_64/onedrive-2.4.25-1.2.x86_64.rpm`](https://download.opensuse.org/repositories/openSUSE:/Factory/standard/x86_64/onedrive-2.4.25-1.2.x86_64.rpm) at `/usr/bin/onedrive` shall affect [`OneDriveGUI-1.0.3-x86_64.AppImage`](https://github.com/bpozdena/OneDriveGUI/releases/download/v1.0.3/OneDriveGUI-1.0.3-x86_64.AppImage) at `$HOME/Applications/OneDriveGUI-1.0.3-x86_64_2be1cb7745a73d7fb999ea3697c5048e.AppImage`, consequently.

	I want to confirm this because, similarly to [`flatpak`](https://github.com/flatpak/flatpak.git) packages, [`github.com/AppImage/AppImageKit`](https://github.com/AppImage/AppImageKit.git)-based `.AppImage` files frequently include their dependencies inside them.

1.	> FYI — from reading a lot of your posts here and elsewhere, I find following your comments / posts / raising items then trying to link back — this all weaves at times a very difficult thread to follow at times. If you could perhaps tweak how you are raising items (issues, discussions and so on) that would be potentially helpful.

	You're, to my knowledge, the first person to mention that, so I've little context to go on — you'll need to be somewhat more specific for that to be actionable. Thanks, though.

**Comment by roke-julian-lockhart** [comment]
This discussion isn't resolved, so I don't believe that it should be closed.

**Reply by roke-julian-lockhart** [reply]
I'd hide this comment, [^1] except for [`orgs/community/discussions/142061`](https://github.com/orgs/community/discussions/142061#discussion-7343560) being unresolved.

[^1]: [`#discussioncomment-15039865`](https://github.com/bpozdena/OneDriveGUI/discussions/137#discussioncomment-15039865)

**Comment by bpozdena** [comment, marked as answer]
The onedrive client will never be bundled with the GUI.

**Reply by roke-julian-lockhart** [reply]
@bpozdena, thanks! Glad to hear.
